src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+Because the Flutter framework is written in Dart, compiling it to JavaScript was
+relatively straightforward.
+
+However, the Flutter engine, written in C++, is designed to interface with the
+underlying operating system rather than a web browser. A different approach is
+therefore required. On the web, Flutter provides a reimplementation of the
+engine on top of standard browser APIs. We currently have two options for
+rendering Flutter content on the web: HTML and WebGL. In HTML mode, Flutter uses
+HTML, CSS, Canvas, and SVG. To render to WebGL, Flutter uses a version of Skia
+compiled to WebAssembly called
+[CanvasKit](https://skia.org/user/modules/canvaskit). While HTML mode offers the
+best code size characteristics, CanvasKit provides the fastest path to the
+browser's graphics stack, and offers somewhat higher graphical fidelity with the
+native mobile targets<sup><a href="#a5">5</a></sup>.
+
+The web version of the architectural layer diagram is as follows:
+
+![Flutter web
+architecture](../../images/arch-overview/web-arch.svg){:width="100%"}
+
+Perhaps the most notable difference compared to other platforms on which Flutter
+runs is that there is no need for Flutter to provide a Dart runtime. Instead,
+the Flutter framework (along with any code you write) is compiled to JavaScript.
+It’s also worthy to note that Dart has very few language semantic differences
+across all its modes (JIT versus AOT, native versus web compilation), and most
+developers will never write a line of code that runs into such a difference.
+
+During development time, Flutter web uses
+[`dartdevc`](https://dart.dev/tools/dartdevc), a compiler that supports incremental compilation and therefore allows hot restart (although not currently hot reload) for apps. Conversely, when you are ready to create a production app for the web, [`dart2js`](https://dart.dev/tools/dart2js), Dart’s highly-optimized
+production JavaScript compiler is used, packaging the Flutter core and framework
+along with your application into a minified source file that can be deployed to
+any web server. Code can be offered in a single file or split into multiple
+files through [deferred
+imports](https://dart.dev/guides/language/language-tour#lazily-loading-a-library).
+
+## Further information
+
+For those interested in more information about the internals of Flutter, the
+[Inside Flutter](https://flutter.dev/docs/resources/inside-flutter) whitepaper
+provides a useful guide to the framework’s design philosophy.
+
+---
+
+**Footnotes:**
+
+<sup><a name="a1">1</a></sup> While the `build` function returns a fresh tree,
+you only need to return something \_different \_if there's some new
+configuration to incorporate. If the configuration is in fact the same, you can
+just return the same widget.
+
+<sup><a name="a2">2</a></sup> This is a slight simplification for ease of
+reading. The tree might in practice be slightly more complex.
```

Comment:
=> In practice, the tree might be more complex.

Reply:
Tim might in theory be able to write a decent sentence. In practice, Tim needs a copy editor to organize his thoughts :)